class/class7.1/src/UnitTests.hpp

```diff
@@ -0,0 +1,119 @@
+#pragma once
+
+#include <vector>
+#include <functional>
+#include <string>
+#include <iostream>
+#include <stdexcept>
+#include <source_location>
+
+
+class Rectangle {
+public:
+    Rectangle(double width, double height);
+    double getWidth() const;
+    double getHeight() const;
+    double getArea() const;
+    bool canFitInside(const Rectangle& other) const;
+
+private:
+    double width_;
+    double height_;
+    double area_;
+};
+
+class RectangleProcessor {
+public:
+    void addRectangle(double width, double height);
+    const Rectangle& getBiggestAreaRectangle() const;
+    const Rectangle& getSmallestAreaRectangle() const;
+    void compareRectangles(size_t index1, size_t index2) const;
+
+private:
+    void updateBiggestAndSmallestArea(double area);
+    std::vector<Rectangle> rectangles_;
+    double biggestArea_ = 0;
+    double smallestArea_ = 1000000;
```

Comment:
double max value can be used instead
Also it is not clear why those classes are in UnitTests header